tasks/config/karma.js

```diff
@@ -49,6 +53,23 @@ module.exports = function (grunt) {
           { type: 'text-summary' },
         ]
       }
-    }
+    },
   };
+
+  times(TOTAL_CI_SHARDS, i => {
+    const n = i + 1;
+    config[`ciShard-${n}`] = {
+      singleRun: true,
+      options: {
+        files: [
+          'http://localhost:5610/bundles/commons.bundle.js',
+          `http://localhost:5610/bundles/tests.bundle.js?shards=${TOTAL_CI_SHARDS}&shard_num=${n}`,
+          'http://localhost:5610/bundles/commons.style.css',
+          'http://localhost:5610/bundles/tests.style.css'
+        ]
+      }
+    };
+  });
+
+  return config;
```

Comment:
I think we should put a high-level description of the problem this PR solves, at the top of the file:

/**
 * -------------------------------------------------------------------------------------------------
 * CI sharding
 * -------------------------------------------------------------------------------------------------
 *
 * Without CI sharding, nearly every test will retain all of the memory it causes to be allocated.
 * This is a problem that will take some serious work to fix. CI sharding is a short-term solution.
 *
 * Rather than modify the bundling process, we'll shard the top-level test suites by name. We can add
 * additional shards by changing `TOTAL_CI_SHARDS`.
 * Sharding is accomplished by murmur3 hashing the top level suite names then bucketing based on the
 * hash output. If a test suite resolves to shard2, but we are running shard1, we simply never pass
 * the function to mocha. Rather than redefine every describe statement, we're intercepting the
 * global window.describe() function to accomplish this.
 */